jdaviz/configs/imviz/helper.py

```diff
@@ -209,6 +209,83 @@ def offset_by(self, dx, dy):
                 viewer.state.x_max = viewer.state.x_min + width
                 viewer.state.y_max = viewer.state.y_min + height
 
+    @property
+    def zoom_level(self):
+        """Zoom level:
+
+        * 1 means real-pixel-size.
+        * 2 means zoomed in by a factor of 2.
+        * 0.5 means zoomed out by a factor of 2.
+        * 'fit' resets the zoom.
+
+        """
+        viewer = self.app.get_viewer("viewer-1")
+        i_top = get_top_layer_index(viewer)
+        image = viewer.layers[i_top].layer
+        nx = image.shape[viewer.state.x_att.axis]
+        ny = image.shape[viewer.state.y_att.axis]
+        zoom_x = nx / (viewer.state.x_max - viewer.state.x_min)
+        zoom_y = ny / (viewer.state.y_max - viewer.state.y_min)
+
+        return max(zoom_x, zoom_y)  # Similar to Ginga get_scale()
+
+    # Loosely based on glue/viewers/image/state.py
+    @zoom_level.setter
+    def zoom_level(self, val):
+        if ((not isinstance(val, (int, float)) and val != 'fit') or
+                (isinstance(val, (int, float)) and val <= 0)):
+            raise ValueError(f'Unsupported zoom level: {val}')
+
+        viewer = self.app.get_viewer("viewer-1")
+        if (viewer.state.reference_data is None or viewer.state.x_att is None
+                or viewer.state.y_att is None):
+            return
+
+        # Want to zoom what is actually visible.
+        # Zoom on X and Y will auto-adjust.
+        i_top = get_top_layer_index(viewer)
```

Comment:
Actually, maybe in this case, unlike contrast/bias, we can just use reference image... right, @astrofrog ?

Reply:
Yes